src/experimental/patterns/ec2-app.ts

```diff
@@ -0,0 +1,307 @@
+ *
+ * @remarks
+ * - The rate at which instances are replaced are conditional based on how scaled up the service is.
+ *   The more an application is scaled, the longer it'll take to deploy.
+ * - It is not necessary to use the `autoscaling` Riff-Raff deployment type in conjunction with this.
+ *   Doing so won't break anything, however it'll lengthen the deployment time as instances will be rotated twice.
+ *   Please remove it from any manually created `riff-raff.yaml` file.
+ * - Your application should include a build number in its filename.
+ *   This value will change across builds, and therefore create a CloudFormation template difference to be deployed.
+ * - Requires the AWS CLI and `cfn-signal` binaries to be available on the instance, and on the `PATH`.
+ *   AMIgo adds these via the `aws-tools` role.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html#cfn-attributes-updatepolicy-rollingupdate
+ * @see https://github.com/guardian/testing-asg-rolling-update
+ *
+ * @experimental
+ */
+export class GuEc2AppExperimental extends GuEc2App {
+  constructor(scope: GuStack, props: GuEc2AppExperimentalProps) {
+    const { minimumInstances, maximumInstances = minimumInstances * 2 } = props.scaling;
+    const { applicationPort } = props;
+    const { region, stackId } = scope;
+
+    super(scope, {
+      ...props,
+      updatePolicy: UpdatePolicy.rollingUpdate({
+        maxBatchSize: maximumInstances,
+        minInstancesInService: minimumInstances,
+        minSuccessPercentage: 100,
+        waitOnResourceSignals: true,
+
+        /*
+        If a scale-in event fires during an `AutoScalingRollingUpdate` operation, the update could fail and rollback.
+        For this reason, we suspend the `AlarmNotification` process, else availability of a service cannot be guaranteed.
+        Consequently, services cannot scale-out during deployments.
+        If AWS ever supports suspending scale-out and scale-in independently, we should allow scale-out.
+         */
+        suspendProcesses: [ScalingProcess.ALARM_NOTIFICATION],
+      }),
+    });
+
+    const { autoScalingGroup, targetGroup } = this;
+    const { userData, role } = autoScalingGroup;
+    const cfnAutoScalingGroup = autoScalingGroup.node.defaultChild as CfnAutoScalingGroup;
+
+    cfnAutoScalingGroup.desiredCapacity = minimumInstances.toString();
+
+    /*
+    Enable all ASG metrics for improved observability.
+    Specifying `granularity` without specifying `metrics` results in all metrics being enabled.
+    See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-metricscollection.html
+     */
+    cfnAutoScalingGroup.metricsCollection = [
+      {
+        granularity: "1Minute",
+      },
+    ];
+
+    // TODO are these sensible values?
+    const signalTimeoutSeconds = Math.max(
+      targetGroup.healthCheck.timeout?.toSeconds() ?? 0,
```

Comment:
I think we could drop this one for simplicity. IIUC it is the timeout _per healthcheck request_, so it will typically be very short. For example, I think we're [defaulting to 5 seconds here](https://github.com/guardian/cdk/blob/12817108d384a561c85f7713a3e943c26fa3a6f3/src/constructs/loadbalancing/alb/application-target-group.ts#L33-L44).